### PR TITLE
Fix templates import and require rio-tiler>=4.1.11

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**

### CODE REVIEW

File: infrastructure/aws/lambda/Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ COPY titiler/ titiler/
 # and becaise we NEED to remove both boto3 and botocore to save space for the package
 # we have to force using old package version that seems `almost` compatible with Lambda env botocore
 # https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
-RUN pip install . "cftime" "mangum>=0.10.0" "pandas==1.5.3" "botocore==1.29.76" "aiobotocore==2.5.0" "s3fs==2023.4.0" "fsspec==2023.4.0" "zarr==2.14.2" "xarray==0.19.0" -t /asset --no-binary pydantic
+RUN pip install . "rio-tiler>=4.1.11" "cftime" "mangum>=0.10.0" "pandas==1.5.3" "botocore==1.29.76" "aiobotocore==2.5.0" "s3fs==2023.4.0" "fsspec==2023.4.0" "zarr==2.14.2" "xarray==0.19.0" -t /asset --no-binary pydantic
```
Comment (vincentsarago) [inline infrastructure/aws/lambda/Dockerfile:18]: there is no rio-tiler 4.1.11

File: titiler/xarray/factory.py [hunk 2]
```diff
@@ -440,7 +440,7 @@ def map_viewer(
                 tilejson_url += f"?{urlencode(request.query_params._list)}"
 
             tms = self.supported_tms.get(TileMatrixSetId)
-            return templates.TemplateResponse(
+            return self.templates.TemplateResponse(
```
Comment (vincentsarago) [inline titiler/xarray/factory.py:443]: thanks for catching this. I did make a small breaking change in the last titiler release 😓 

I fixed this in https://github.com/developmentseed/titiler-xarray/commit/b9660164c9154ec8f30ef990f76effba3488caa2

File: infrastructure/aws/lambda/Dockerfile (hunk 1 above)
Comment (vincentsarago) [reply infrastructure/aws/lambda/Dockerfile:18]: 🤦 https://github.com/cogeotiff/rio-tiler/blob/main/CHANGES.md#4111-2023-05-18

**Comment by abarciauskas-bgse** [comment]
Noted that the templates are fixed, so I merged main and this PR is just adding the new rio-tiler version.

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**